dashboard/signals.py

```diff
@@ -25,27 +38,61 @@ def precreate_programenrollment(sender, instance, **kwargs):  # pylint: disable=
         instance_in_db = CachedEnrollment.objects.filter(id=instance.id).exclude(data__isnull=True).count()
         # if the count is 1, it means the student unenrolled from the course run
         if instance_in_db == 1:
-            active_enrollment_count = CachedEnrollment.objects.filter(
-                user=user,
-                course_run__course__program=program
-            ).exclude(data__isnull=True).count()
             # if there is only one enrollment with data non None, it means that it is the
             # current instance is the only one for the program, so the program enrollment
             # needs to be deleted
-            if active_enrollment_count <= 1:  # theoretically this cannot be <1, but just in case
+            if CachedEnrollment.active_count(user, program) <= 1:  # theoretically this cannot be <1, but just in case
                 ProgramEnrollment.objects.filter(
                     user=user,
                     program=program
                 ).delete()
 
 
-@receiver(post_save, sender=CachedEnrollment, dispatch_uid="update_programenrollment")
-def create_programenrollment(sender, instance, **kwargs):  # pylint: disable=unused-argument
+@receiver(post_save, sender=CachedEnrollment, dispatch_uid="cachedenrollment_post_save")
+def handle_update_enrollment(sender, instance, **kwargs):  # pylint: disable=unused-argument
```

Comment:
it's doing a few different things now. it's creating (or getting) a program enrollment, and it's reindexing based on that enrollment. it's not just creating a ProgramEnrollment anymore, so it makes sense to change the name